tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -987,13 +1058,15 @@ private[emitter] final class ClassEmitter(jsGen: JSGen) {
     val (createNamespace, expAccessorVar) =
       genCreateNamespaceInExports(fullName)
 
-    val methodDef = desugarToFunction(cd.encodedName, args, body,
+    val methodDefWithGlobals = desugarToFunction(cd.encodedName, args, body,
         isStat = resultType == NoType)
 
-    js.Block(
-        createNamespace,
-        expAccessorVar := methodDef
-    )
+    for (methodDef <- methodDefWithGlobals) yield {
+      js.Block(
+          createNamespace,
+          expAccessorVar := methodDef
+      )
+    }
```

Comment:
Yes. IIUC, `ClassEmitter` "doesn't actually change" in this commit. So if we can put the wiring in another one, we get:

- Guarantee that "nothing changes".
- Less load for someone actually looking at this commit.